hubploy/auth.py

```diff
@@ -43,6 +47,39 @@ def registry_auth_gcloud(deployment, project, service_key):
     ])
 
 
+def registry_auth_aws(deployment, project, service_key):
+    """
+    Setup AWS authentication with a service_key
+
+    This changes *global machine state* on where docker can push to!
+    """
+    import boto3
+    
+    service_key_path = os.path.join(
+        'deployments', deployment, 'secrets', service_key
+    )
+
+    if not os.path.isfile(service_key_path):
+        raise FileNotFoundError(f'The service_key file {service_key_path} does not exist')
+
+    os.environ['AWS_SHARED_CREDENTIALS_FILE'] = os.path.abspath(service_key_path)
+
+    client = boto3.client(
+        'ecr'
+    )
+    response = client.get_authorization_token()
+    auth_dict = response['authorizationData'][0]
+
+    authorizationToken = auth_dict['authorizationToken']
+    proxyEndpoint = auth_dict['proxyEndpoint']
+
+    subprocess.check_call([
+      'docker', 'login', '-u', 'AWS',
```

Comment:
In https://circleci.com/gh/pangeo-data/pangeo-cloud-federation/163, we're seeing a failure in this line: 

```
Traceback (most recent call last):
  File "/root/repo/venv/bin/hubploy", line 11, in <module>
    load_entry_point('hubploy==0.1.0', 'console_scripts', 'hubploy')()
  File "/root/repo/venv/lib/python3.6/site-packages/hubploy/__main__.py", line 53, in main
    auth.registry_auth(args.deployment)
  File "/root/repo/venv/lib/python3.6/site-packages/hubploy/auth.py", line 24, in registry_auth
    deployment, **registry['aws']
  File "/root/repo/venv/lib/python3.6/site-packages/hubploy/auth.py", line 79, in registry_auth_aws
    proxyEndpoint
  File "/usr/lib/python3.6/subprocess.py", line 286, in check_call
    retcode = call(*popenargs, **kwargs)
  File "/usr/lib/python3.6/subprocess.py", line 267, in call
    with Popen(*popenargs, **kwargs) as p:
  File "/usr/lib/python3.6/subprocess.py", line 709, in __init__
    restore_signals, start_new_session)
  File "/usr/lib/python3.6/subprocess.py", line 1344, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'docker': 'docker'
```

cc @yuvipanda and @apawloski